File: vaxsul_frontend/src/types/purchase.ts
import { z } from "zod";

export const PurchaseSchema = z.object({
  id: z.number().optional().nullable(),
  vaccineId: z.number(),
  userId: z.number().optional().nullable(),
  amount: z.number(),
  totalSpent: z.number(),
  timestamp: z.string(),
});

export type Purchase = z.infer<typeof PurchaseSchema>;

File: vaxsul_backend/src/main/kotlin/com/github/dannful/domain/model/Purchase.kt
package com.github.dannful.domain.model

import kotlinx.datetime.LocalDateTime
import kotlinx.serialization.Serializable

@Serializable
data class Purchase(
    val userId: Int,
    val vaccineId: Int,
    val totalSpent: Float,
    val amount: Int,
    val timestamp: LocalDateTime? = null,
    val status: PurchaseStatus
)

@Serializable
data class IdPurchase(
    val id: Int,
    val userId: Int,
    val vaccineId: Int,
    val totalSpent: Float,
    val amount: Int,
    val timestamp: LocalDateTime? = null,
    val status: PurchaseStatus
)
